app/lib/service/openid/openssl_commands.dart

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+import 'dart:typed_data';
+
+import 'package:meta/meta.dart';
+import 'package:pana/pana.dart';
+import 'package:path/path.dart' as p;
+import 'package:pem/pem.dart';
+import 'package:pub_dev/shared/utils.dart';
+
+/// Randomly generated private and public RSA keypair.
+@visibleForTesting
+class RsaKeyPair {
+  final Uint8List privateKey;
+  final Uint8List publicKey;
+
+  RsaKeyPair(this.privateKey, this.publicKey);
+}
+
+/// Generates a new RSA keypair using `openssl`.
+@visibleForTesting
+Future<RsaKeyPair> generateRsaKeyPair({
```

Comment:
We don't strictly need this, can't we just hardcode a few key public keys into the test cases? :see_no_evil:

Reply:
And that also counts for the signing part. Let's keep this as small as possible.

Reply:
Moved these to the `test/` subdirectory, because I think they may be needed in the near future for integration tests. Added a TODO to review/remove them later.